docs/DeveloperGuide.md

```diff
@@ -2,12 +2,111 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+We first give acknowledgement to Module Coordinator, Dr Akshay Narayan, and Teaching Assistant, Aditi Chadha for guidance and supervision over the KaChinnnng project.
+
+We utilised the following resources to aid us in the development of KaChinnnng:
+-  [tp project](https://github.com/nus-cs2113-AY2324S1/tp) of the nus-cs2113-AY2324S1 organisation
+-  [addressbook-level2](https://github.com/se-edu/addressbook-level2) & [addressbook-level3](https://github.com/se-edu/addressbook-level3) project by SE-EDU largely inspired the format and coding style of KaChinnnng.
+
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Below are the design and implementation details of KaChinnnng.
+
+### Design
+
+The architectural diagram of KaChinnnng is as follows:
+
+![ArchitectureDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/ArchitectureDiagram.png?raw=true)
+
+The kaChinnnng program will first enter the `run` state where the `storage` class will be accessed and data will be retrieved if the application has been used before.
+
+Next, the `User` will input various `commands` via the `Ui` class. The `commands` will then be parsed by the `Parser` class and the relevant `managers` will be called to execute the `commands`.
+
+Once the `User` inputs has been successfully executed, `commands` will make use of the `Ui` class to output the relevant information to the user.
+
+When the `User` exits the program, the `storage` class will be called to save the data to the local storage.
+
+`Storage` will read from the local storage when the program is run again.
+
+### Expense Class Overview
+
+The Expenses are divided into three categories. Transport, Utilities and Food.
+Each of the categories inherit from a base Expense class which in turn
+inherits from the FinancialRecord class.
+
+![expense_class_diagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/expense_class_diagram.png?raw=true)
```

Comment:
Do you have a key for the visibility of each method and attribute in your classes? It's not entirely clear which access modifiers each class' methods and attributes use.